daemon/graphdriver/overlay2/overlay.go

```diff
@@ -688,9 +687,12 @@ func (d *Driver) ApplyDiff(id string, parent string, diff io.Reader) (size int64
 	}
 
 	if d.options.syncDiffs {
-		// FIXME: Instead of syncing all the filesystems we should be fsyncing each
-		// file as the tar archive gets unpacked
```

Comment:
I don't know the big picture, so maybe this is irrelevant, but the PR is not doing exactly the same as this comment suggested to do. I mean, if the filesystem was seeing a lot of activity beyond the diff-applying, this new syncing code could still be doing much more that we need, right?

I see this is an improvement, but maybe the `FIXME` comment is still relevant.

Reply:
I wonder if that comment is talking about putting the sync call here: https://github.com/balena-os/balena-engine/blob/master/pkg/archive/archive.go#L1003-L1005 ?

I don't like how deep of a change into the unpacking logic that would be (and that we won't have an easy way to skip it like we do here)

Reply:
I don't see the utility of syncing after every extracted file compared to syncing the target filesystem after the complete archive is extracted. Syncing individual files stops a file from being partially written, but doesn't stop the entire diff from being partially written, which is just as bad.

Reply:
doesn't a `syncfs` on the data partition lead to also flushing writes that a user app could be doing on a volume or the container rootfs. this is already making the scope of the sync smaller, so that's fine in my book :) 

I think what this comment was suggesting is that in the ideal case you would only force sync writes we caused in the graphdriver.

Reply:
> doesn't a syncfs on the data partition lead to also flushing writes that a user app could be doing on a volume or the container rootfs

Good wording! That's exactly what I had in mind in my comment when I said "if the filesystem was seeing a lot of activity beyond the diff-applying, this new syncing code could still be doing much more that we need."